flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/ExprCodeGenerator.scala

```diff
@@ -476,6 +485,11 @@ class ExprCodeGenerator(ctx: CodeGeneratorContext, nullableInput: Boolean)
             call.getOperator.getReturnTypeInference == ReturnTypes.ARG0 =>
         generateNullLiteral(resultType)
 
+      // We only support JSON function operands within JSON_OBJECT
+      case (operand: RexNode, _)
+          if isJsonObjectOrArrayOperand(call) && isJsonFunctionOperand(operand) =>
```

Comment:
Since we use `OrArrayOperand`? Do we support JSON_ARRAY?

Reply:
I was reusing a method here and updated it to check specifically for json object. The intent of the PR is to add support only for json object. If we want to extend it, we can do it in other tickets/pr.